analyses/molecular-subtyping-ATRT/01-ATRT-molecular-subtyping-data-prep.Rmd

```diff
@@ -241,50 +254,60 @@ long_stranded_expression <- scale(t(filtered_expression),
 # Merge metadata with expression data
 expression_metadata <- long_stranded_expression %>%
   as.data.frame() %>%
-  tibble::rownames_to_column("Kids_First_Biospecimen_ID") %>%
-  dplyr::left_join(select_metadata, by = "Kids_First_Biospecimen_ID")
+  tibble::rownames_to_column("Kids_First_Biospecimen_ID") %>% 
+  dplyr::left_join(select_metadata, by = "Kids_First_Biospecimen_ID") %>%
+  dplyr::select(-Kids_First_Biospecimen_ID)
 
 # Display `expression_metadata`
 expression_metadata %>%
   head(n = 15)
 
 # Join expression data with metadata filtered for `ATRT`
-atrt_expression_df <- metadata %>%
+atrt_expression_df <- collapsed_metadata %>%
   dplyr::left_join(expression_metadata,
                    by = "sample_id")
+
+# Remove data we no longer need
+rm(filtered_expression, long_stranded_expression, expression_metadata)
 ```
 
 ### CN data
 
 ```{r}
-# Filter focal CN data for SMARCB1 and SMARCA4 status 
-cn_df <- cn_df %>% 
+# Filter focal CN data for SMARCB1 and SMARCA4 status
+genes_cn_df <- cn_df %>%
   dplyr::filter(gene_symbol %in% c("SMARCB1", "SMARCA4")) %>%
-  dplyr::mutate(
-    SMARCB1_focal_status = dplyr::case_when(gene_symbol == "SMARCB1" ~ status,
-                                            TRUE ~ "neutral"),
-    SMARCA4_focal_status = dplyr::case_when(gene_symbol == "SMARCA4" ~ status,
-                                            TRUE ~ "neutral")
-  ) %>%
-  dplyr::select(-c("status", "gene_symbol")) %>%
-  dplyr::distinct() %>%
-  dplyr::group_by(sample_id) %>%
-  dplyr::summarise(
-    SMARCB1_focal_status = paste(sort(unique(
-      SMARCB1_focal_status
-    )), collapse = ", "),
-    SMARCA4_focal_status = paste(sort(unique(
-      SMARCA4_focal_status
-    )), collapse = ", ")
-  )
-
-# Display `cn_metadata`
-cn_df %>%
+  tidyr::spread(gene_symbol, status, fill = "neutral") %>%
+  dplyr::rename(SMARCB1_focal_status = SMARCB1,
+                SMARCA4_focal_status = SMARCA4)
+
+# add in the samples that did not have any copy number changes for either
```

Comment:
Before, if a biospecimen did not have evidence for an alteration in either _SMARCB1_ or _SMARCA4_, it would look like it was "missing" in the final data.frame. This is likely because we filter out neutral events here: 

https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/2cbf526e24d92c11f093b33aa31663546b3fbb57/analyses/focal-cn-file-preparation/01-prepare-cn-file.R#L262

to save on memory for the annotation step and disk space when we save the prepped file. So here, we assume that if a sample appears _anywhere_ in the CNVkit file that we did assay it but that there was no evidence of a copy number change so we mark it neutral.

@jharenza you should be aware of this assumption as you look at the table! Not sure how common it is to essentially 'miss' segments when calling copy number from WGS and therefore _not_ assay  _SMARCB1_ and _SMARCA4_ for those samples...

Reply:
thanks for the heads up! The normal output from segmentation algorithms don't contain neutral segments, so this makes sense.